docs/DeveloperGuide.md

```diff
@@ -182,6 +156,42 @@ More details about the reference frame for executing the done command is shown b
 
 <img src="images/DoneCryptoExecuteDiagram.png" width="600"/>
 
+### Ui Component
+
+The ui component only contains the TextUi.java file and its API can be found
+[here](https://github.com/AY2122S1-CS2113T-T12-1/tp/blob/master/src/main/java/seedu/mtracker/ui/TextUi.java).
+
+It is a basic java class containing string attributes and helper methods for displaying the different features, texts and
+instructions to the user. Hence, **under the single-responsibility principle (SRP), its only responsibility is to act as the primary interaction platform
+between the user and the rest of the program**.
+
+As detailed by the UML diagrams in the Architecture sections above, **many other parser and command classes utilize
+the methods contained in `TextUi`** to display instructions on the console for required user input. Hence, **most other
+classes of this program are dependent on the methods of this `TextUi` class** for their proper interaction with the user.
+
+Thus, the **`TextUi` class is highly-cohesive** as it contains all the user text display methods for the various classes
+in itself. This **enhances maintainability** as only this class **has to be modified to achieve a small change in
+the desired texted or instruction to be displayed by various classes**, and **increases reusability of the module**
+as all aspects of texts or instruments to be displayed on the console **have been localized**.
+
+On the other hand, the `TextUi` class itself **has a dependency only on an Instrument class** whenever
+the user wishes to `list` out all the instruments in the watchlist or if s/he wants to `view`
+one such instrument in detail. The following sequence diagram explains `TextUi`'s interaction with an `Instrument` class when
+ListCommand.execute() calls the `displayInstrument()` method when the user wishes to list out all instruments in the watchlist:
+
+<img src="images/TextUiDisplayInstrumentGeneralParams.png" width="780"/>
```

Comment:
I dont think need the first note cause in the text u alr said that execute invokes the displayAllInstruments()
The note dont need Instrument.getGeneralParams(), just say "The getGeneralParams() method ...